packages/tldraw/src/lib/shapes/text/toolStates/Pointing.ts

```diff
@@ -168,12 +168,22 @@ export class Pointing extends StateNode {
 			const transform = this.editor.getShapeParentTransform(shape)
 			delta.rot(-transform.rotation())
 		}
-
-		this.editor.updateShape({
-			...shape,
-			x: shape.x + delta.x,
-			y: shape.y + delta.y,
-		})
+		const isGridMode = this.editor.getInstanceState().isGridMode
+		if (isGridMode) {
+			const gridSize = this.editor.getDocumentSettings().gridSize
+			const newPoint = new Vec(shape.x, shape.y).snapToGrid(gridSize)
```

Comment:
this can use your `maybeSnapToGrid` fn, right?

Reply:
I didn't think so, because it would strip the delta. But now I reckon I can actually.

Reply:
I could either:

1. Check if we're in grid mode, then call maybeSnapToGrid, which would check gridMode again.
2. Change maybeSnapToGrid so that it handles text in a special way, or returns a bool to indicate whether it snapped or not
3. Leave as is, and duplicate the logic

Personally I prefer keeping the function simple, since it works fine everywhere else, and simply duplicating the logic, rather than adding the extra complexity. What do you think?